backend/matchAlgorithm.js

```diff
@@ -1,40 +1,143 @@
-import prisma from './prisma.js';
+import { PrismaClient } from '@prisma/client';
+import { fetchCosineSimilarity } from './TextSimilarity.js';
 
+const prisma = new PrismaClient();
 
-async function galeShapley() {
+export async function galeShapley(loggedInUserId, userRole) {
   const adopters = await prisma.adopter.findMany();
   const adoptees = await prisma.adoptee.findMany();
+  const rawAdopterPreferences = await prisma.adopterPreference.findMany();
```

Comment:
This fetch all adopters and adoptees including their preferences(rank) from the database